validation/validation.go

```diff
@@ -33,39 +31,56 @@ func ValidatePort(port string, field string) ErrorList {
 	return allErrs
 }
 
-// ValidateProtocol validates that given value belongs to supported protocols
-func ValidateProtocol(protocol string, field string) ErrorList {
+// ValidatePortRange validates that the given value is a valid port
+// range, with its elements in range 0 - 65535.  It accepts singular
+// ports P, and port ranges of the form N-M.
+func ValidatePortRange(portrange string, field string) ErrorList {
 	allErrs := ErrorList{}
 
-	if err := IsValidProtocol(protocol); err != nil {
-		allErrs = append(allErrs, NotSupported(field, protocol, []string{TCP, UDP}))
+	// Syntax:
+	// % int
+	// % int-int
 
+	matches := patternPortSingular.FindStringSubmatch(portrange)
+	if len(matches) == 0 {
+		matches = patternPortRange.FindStringSubmatch(portrange)
+		if len(matches) == 0 {
+			return append(allErrs, Invalid(field, portrange, `invalid syntax`))
+		}
+	}
+	// Note: matches[0] is the fist/left-most match. Here this is
+	// entire string, if it matched, due to the ^...$ bracketing.
+	// The captures, we only part we are interested in, start at
```

Comment:
Now I wonder what my (compose) is. Well, just a curiosity, focus, tests more important.